src/icons/Icon.scss

```diff
@@ -0,0 +1,43 @@
+.icon {
+  --icon-utility-sm: 0.75rem;
+  --icon-utility-sm-alignment: -0.05rem;
+  --icon-utility-md: 1rem;
+  --icon-utility-md-alignment: -0.125rem;
+  --icon-badge-lg: 2.5rem;
+  --icon-badge-lg-alignment: -0.75rem;
+  --icon-badge-xl: 3rem;
+  --icon-badge-xl-alignment: -1rem;
+  --icon-height: 1em;
+  --icon-alignment: -0.125em;
+
+  display: inline-block;
+  vertical-align: var(--icon-alignment);
+  height: var(--icon-height);
+
+  &[data-size="utility-sm"] {
+    --icon-height: var(--icon-utility-sm);
+    --icon-alignment: var(--icon-utility-sm-alignment);
+  }
+
+  &[data-size="utility-md"] {
+    --icon-height: var(--icon-utility-md);
+    --icon-alignment: var(--icon-utility-md-alignment);
+  }
+
+  &[data-size="badge-lg"] {
+    --icon-height: var(--icon-badge-lg);
+    --icon-alignment: var(--icon-badge-lg-alignment);
+  }
+
+  &[data-size="badge-xl"] {
+    --icon-height: var(--icon-badge-xl);
+    --icon-alignment: var(--icon-badge-xl-alignment);
+  }
+}
+
+.icon svg {
+  display: block;
+  height: 100%;
+  aspect-ratio: 1 / 1;
+  fill: currentColor;
```

Comment:
Where does the value `currentColor` come from?

Reply:
It's a special keyword (is that the right term ?) in CSS which means just pull in whatever the text color is. That's how we can automatically get a blue icon with blue text, red icon with red text, etc.